File: Прогр1J/src/WaterTap.java
public class WaterTap {
	private boolean open;

	public boolean GetOpen() {
		return open;

	}

	public void SetOpen(boolean o) {
		open = o;
	}

	public void Wash(Egg egg) {
		if (open) {
			egg.SetDirty(false);
		}
	}
}

File: Прогр1(1)/Прогр1(1)/WaterTap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Прогр1_1_
{
    class WaterTap
    {

        private bool open { set; get; }
        public bool Open { set { open = value; } get { return open; } }


        public void Wash(Egg egg)
        {
            if (open)
            {
                egg.Dirty = false;
                Console.WriteLine("Яйцо вымыто");
            }
        }
    }
}
